libraries/EEPROM/src/utility/stm32_eeprom.c

```diff
@@ -175,6 +198,8 @@ void eeprom_buffered_write_byte(uint32_t pos, uint8_t value)
   */
 void eeprom_buffer_fill(void)
 {
+  if(is_buffer_filled)
+    return;
```

Comment:
Don't think it is a good idea. User should be able to perfom a `eeprom_buffer_fill` when he wants. See the `BufferedEEPROM` example. Moreover it It guarantees the buffer is well aligned with the flash.
```suggestion
```

Reply:
I agree, filling the buffer again is a way to discard the changes before they are written to the flash.
But at the same time I feel that `EEPROMClass::read` doing that on each byte read is a pessimization. It's true that user can read data using `eeprom_buffered_read_byte` if he doesn't want that, but shouldn't the library offer the same capability through `EEPROMClass`?